### PR TITLE
[tests-only][full-ci] removing the setresponse in given/then step in AppConfigurationContext and AuthContext

### PR DESCRIPTION
**Pull request by KarunAtreya**
## Description
We have used setResponse() and $this->response in the Given/Then steps and some helper functions (maybe to reuse existing available methods). But storing responses from Given/Then steps and helper functions is not a good idea because it can lead to a false positive assertion in the Then steps.
So, check the use of setResponse() and $this->response in
- Given steps
- Then steps (Then steps can use $this->response but must prevent saving to it)
- Helper functions

So this pr make the above changes in `AppConfigurationContext` and `ArchiverContext`
## Related Issue
https://github.com/owncloud/ocis/issues/7082

## Motivation and Context
- To  remove setResponse() and $this->response in the Given/Then steps and some helper functions
- To avoid false positive assertions

## How Has This Been Tested?

- test environment:
- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/AuthContext.php [hunk 1]
```diff
@@ -126,8 +129,8 @@ public function userRequestsURLWithNoAuth(string $user, string $url, string $met
 	 * @return void
 	 */
 	public function userHasRequestedURLWith(string $url, string $method):void {
-		$this->sendRequest($url, $method);
-		$this->featureContext->theHTTPStatusCodeShouldBeSuccess();
+		$response = $this->sendRequest($url, $method);
+		$this->featureContext->theHTTPStatusCodeShouldBeBetween(200, 209, $response);
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/AuthContext.php:133]:
```suggestion
		$this->featureContext->theHTTPStatusCodeShouldBeBetween(200, 299, $response);
```

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/bootstrap/AuthContext.php [hunk 2]
```diff
@@ -884,11 +887,12 @@ public function administratorRequestsURLWithUsingBasicAuth(string $url, string $
 	 */
 	public function userRequestsURLWithUsingBasicTokenAuth(string $user, string $url, string $method):void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/AuthContext.php:888]: NOTE: This function is used in the Given steps below which should be prevented. But do it in the next iteration. See the second point todo list (https://github.com/owncloud/ocis/issues/7082). First, fix all `setResponse` usage and then check the reuse of step-def functions in other steps and refactor it.

**Review by saw-jan — COMMENTED** [review]
LGTM

File: tests/acceptance/features/bootstrap/AppConfigurationContext.php [hunk 3]
```diff
@@ -110,32 +111,31 @@ public function getAppParameter(string $capabilitiesApp, string $capabilitiesPat
 	 */
 	public function userRetrievesCapabilities(string $username): void {
 		$user = $this->featureContext->getActualUsername($username);
-		$this->userGetsCapabilities($user, true);
+		$response = $this->userGetsCapabilities($user, true);
+		$this->featureContext->setResponse($response);
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/AppConfigurationContext.php:115]: let's make this inline
```suggestion
		$this->featureContext->setResponse($this->userGetsCapabilities($user, true));
```

File: tests/acceptance/features/bootstrap/AppConfigurationContext.php [hunk 4]
```diff
@@ -148,14 +148,8 @@ public function userGetsCapabilities(string $username, ?bool $formatJson = false
...
-		$this->userGetsCapabilities($username);
-		$statusCode = $this->featureContext->getResponse()->getStatusCode();
-		if ($statusCode !== 200) {
-			throw new \Exception(
-				__METHOD__
-				. " user $username returned unexpected status $statusCode"
-			);
-		}
+		$response = $this->userGetsCapabilities($username);
+		$this->featureContext->theHTTPStatusCodeShouldBe(200, '', $response);
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/AppConfigurationContext.php:152]: same here
```suggestion
		$this->featureContext->theHTTPStatusCodeShouldBe(200, '', $this->userGetsCapabilities($username));
```

File: tests/acceptance/features/bootstrap/FeatureContext.php [hunk 5]
```diff
@@ -2302,9 +2302,11 @@ public function statusPhpRespondedShouldMatch(PyStringNode $jsonExpected): void
 		$jsonExpectedDecoded = \json_decode($jsonExpected->getRaw(), true);
 		$jsonRespondedDecoded = $this->getJsonDecodedResponse();
 
-		$this->appConfigurationContext->theAdministratorGetsCapabilitiesCheckResponse();
+		$response = $this->appConfigurationContext->userGetsCapabilities($this->appConfigurationContext->getAdminUsernameForCapabilitiesCheck());
+		$this->theHTTPStatusCodeShouldBe(200, '', $response);
+		$response_xml = $this->getResponseXml($response)->data->capabilities;
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/FeatureContext.php:2307]: can we not use snake case as rest of the files are using camel case
```suggestion
		$responseXml = $this->getResponseXml($response)->data->capabilities;
```

**Review by SwikritiT — APPROVED** [review]
LGTM

**Pull request merged**